tests/test_ui_complex_scenarios.py

```
@@ -0,0 +1,841 @@
+                await mock_btn_callbacks['folder_open']()
+
+    asyncio.run(run_file_picker())
+    assert state.get_data_by_path("root")["path_val"] == "new/path.txt"
+
+    path_cb = mock_input_callbacks['path_val']
+    path_cb(MagicMock(value='typed/path.txt'))
+    assert state.get_data_by_path("root")["path_val"] == "typed/path.txt"
+
+    bool_cb = mock_input_callbacks['bool_val']
+    bool_cb(MagicMock(value=False))
+    assert state.get_data_by_path("root")["bool_val"] == False
```

Ruff (0.15.18)

[error] Avoid equality comparisons to `False`; use `not state.get_data_by_path("root")["bool_val"]:` for false checks

Replace with `not state.get_data_by_path("root")["bool_val"]`

(E712)